sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -582,12 +582,26 @@ def _resolve_ir_placeholders_v2(
             'Input argument supports only the following types: PipelineParam'
             ', str, int, float. Got: "{}".'.format(argument_value))
 
+  component_spec = copy.deepcopy(component_spec)
+
   if not component_spec.name:
     component_spec.name = _components._default_component_name
 
   # task.name is unique at this point.
   pipeline_task_spec.task_info.name = (dsl_utils.sanitize_task_name(task.name))
 
+  # IR only supports constant arguments for parameter inputs.
+  # Converting inputs that have constant arguments to parameter inputs.
+  inputs_with_constant_arguments = {
+    name
+    for name, argument in pipeline_task_spec.inputs.parameters.items()
+    if argument.runtime_value.HasField("constant_value")
+  }
+  for input_spec in component_spec.inputs or []:
+    if (input_spec.name in inputs_with_constant_arguments and
+        not type_utils.is_parameter_type(input_spec.type)):
```

Comment:
Looking at the existing implementation, I think this HasField("constant_value") would be true only when the input is already categorized as a parameter. 
Unless I missed something, there won't be a case when this if statement returns true.

Reply:
You're right. I have not realized that the previous code block was only setting `constant_value` when input is already a parameter.
The proper solution is likely closer to https://github.com/kubeflow/pipelines/pull/5478